spark/src/test/scala/org/apache/spark/sql/delta/ColumnMappingUsageTrackingSuite.scala

```diff
@@ -0,0 +1,227 @@
+import org.apache.spark.sql.delta.test.DeltaSQLCommandTest
+
+class ColumnMappingUsageTrackingSuite extends QueryTest with DeltaSQLCommandTest {
+  import testImplicits._
+
+  private val COLUMN_MAPPING_USAGE_TRACKING_FEATURE_KEY =
+    TableFeatureProtocolUtils.propertyKey(ColumnMappingUsageTrackingTableFeature)
+
+  private val COLUMN_MAPPING_USAGE_TRACKING_FEATURE_DEFAULT_KEY =
+    TableFeatureProtocolUtils.defaultPropertyKey(ColumnMappingUsageTrackingTableFeature)
+
+  private val TABLE_NAME = "t"
+
+  test("add column without dropping or renaming column") {
+    withSQLConf(DeltaConfigs.COLUMN_MAPPING_MODE.defaultTablePropertyKey -> "name",
+      COLUMN_MAPPING_USAGE_TRACKING_FEATURE_DEFAULT_KEY -> "supported") {
+
+      withTable(TABLE_NAME) {
+        sql(s"CREATE TABLE $TABLE_NAME (id INT, name STRING) USING delta")
+        assert(!hasDroppedOrRenamedColumn())
+        assert(getPhysicalColumnName("id") === "id")
+        assert(getPhysicalColumnName("name") === "name")
+
+        sql(s"ALTER TABLE $TABLE_NAME ADD COLUMN last_name STRING")
+        assert(!hasDroppedOrRenamedColumn())
+        assert(getPhysicalColumnName("last_name") === "last_name")
+      }
+    }
+  }
+
+  test("add column after dropping column") {
+    withSQLConf(DeltaConfigs.COLUMN_MAPPING_MODE.defaultTablePropertyKey -> "name",
+      COLUMN_MAPPING_USAGE_TRACKING_FEATURE_DEFAULT_KEY -> "supported") {
+
+      withTable(TABLE_NAME) {
+        sql(s"CREATE TABLE $TABLE_NAME (id INT, name STRING) USING delta")
+        assert(!hasDroppedOrRenamedColumn())
+        assert(getPhysicalColumnName("id") === "id")
+        assert(getPhysicalColumnName("name") === "name")
+
+        sql(s"ALTER TABLE $TABLE_NAME DROP COLUMN id")
+        assert(hasDroppedOrRenamedColumn())
+
+        sql(s"ALTER TABLE $TABLE_NAME ADD COLUMN last_name STRING")
+        assert(hasDroppedOrRenamedColumn())
+        assert(getPhysicalColumnName("last_name") !== "last_name")
+      }
+    }
+  }
+
+  test("add column after renaming column") {
+    withSQLConf(DeltaConfigs.COLUMN_MAPPING_MODE.defaultTablePropertyKey -> "name",
+      COLUMN_MAPPING_USAGE_TRACKING_FEATURE_DEFAULT_KEY -> "supported") {
+
+      withTable(TABLE_NAME) {
+        sql(s"CREATE TABLE $TABLE_NAME (id INT, name STRING) USING delta")
+        assert(!hasDroppedOrRenamedColumn())
+        assert(getPhysicalColumnName("id") === "id")
+        assert(getPhysicalColumnName("name") === "name")
+
+        sql(s"ALTER TABLE $TABLE_NAME RENAME COLUMN name TO first_name")
+        assert(hasDroppedOrRenamedColumn())
+        assert(getPhysicalColumnName("first_name") === "name")
+
+        sql(s"ALTER TABLE $TABLE_NAME ADD COLUMN last_name STRING")
+        assert(hasDroppedOrRenamedColumn())
+        assert(getPhysicalColumnName("last_name") !== "last_name")
+      }
+    }
+  }
+
+  test("create table with invalid column names") {
+    withSQLConf(DeltaConfigs.COLUMN_MAPPING_MODE.defaultTablePropertyKey -> "name",
```

Comment:
nit: these two configs are the most common for all tests, maybe make them the default and set the negative explicitly.